src/app/components/ResultProducts/ResultProducts.tsx

```diff
@@ -1,26 +1,49 @@
 import React from 'react';
+import { Product } from '../../../types';
+import SaveIcon from '../Icons/SaveIcon';
 import styles from './ResultProducts.module.css';
 
-type ResultProps = {
-  imgSrc: string;
-  title: string;
-  vendor: string;
-  price: string;
+type ProductProps = {
+  product: Product;
 };
 
-function ResultProducts({
-  imgSrc,
-  title,
-  vendor,
-  price,
-}: ResultProps): JSX.Element {
+function ResultProducts({ product }: ProductProps): JSX.Element {
+  async function postProduct() {
+    const response = await fetch('/api/products', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        ...product,
+        targetPrice: '200€',
+      }),
+    });
+    if (!response.ok) {
+      const errorMessage = await response.text();
+      throw errorMessage;
+    }
+  }
   return (
     <div className={styles.resultProduct}>
-      <img className={styles.resultProduct__img} src={imgSrc} />
+      <img className={styles.resultProduct__img} src={product.image} />
+
       <div className={styles.resultProduct__card}>
-        <h3 className={styles.resultProduct__title}>{title}</h3>
-        <p className={styles.resultProduct__vendor}>{vendor}</p>
-        <p className={styles.resultProduct__price}>{price}</p>
+        <div className={styles.resultProduct__titleWrapper}>
+          <p className={styles.resultProduct__title}>{product.title}</p>
+          <div onClick={() => postProduct()}>
```

Comment:
It's not recommended to use `div` here, better use `button`. You can remove the default styling afterwards.
And you can directly bypass the function instead of creating a new arrow function.

```suggestion
          <button onClick={postProduct}>
```

Reply:
Yes sure thanks!